utils/src/main/java/uk/gov/pay/commons/utils/xray/XRaySessionProfiler.java

```diff
@@ -37,36 +38,15 @@ public void endOperationProfile(String operationName, DatabaseQuery databaseQuer
     @Override
     public Object profileExecutionOfQuery(DatabaseQuery databaseQuery, Record record, AbstractSession abstractSession) {
 
-        HashMap additionalParams = new HashMap();
-        DatabaseMetaData metadata;
-        String hostname = "database";
-        try {
-            Connection connection = abstractSession.getAccessor().getConnection();
-            metadata = connection.getMetaData();
-            additionalParams.put("url", metadata.getURL());
-            additionalParams.put("user", metadata.getUserName());
-            additionalParams.put("driver_version", metadata.getDriverVersion());
-            additionalParams.put("database_type", metadata.getDatabaseProductName());
-            additionalParams.put("database_version", metadata.getDatabaseProductVersion());
-            additionalParams.put("preparation", databaseQuery.isCallQuery() ? "call" : "statement");
-            additionalParams.put("sanitized_query", StringUtils.isEmpty(databaseQuery.getSQLString()) ? "" : databaseQuery.getSQLString());
-
-            try {
-                hostname = new URI((new URI(metadata.getURL())).getSchemeSpecificPart()).getHost();
-                hostname = connection.getCatalog() + "@" + hostname;
-            } catch (URISyntaxException exception) {
-                logger.warn("Error parsing database host name.");
-            }
-        } catch (SQLException exception) {
-            logger.warn("Error getting database connection details.");
-        }
+        HashMap<String, Object> databaseMetadata = new HashMap<>();
+        String hostname = getDatabaseHostnameAndMetadata(databaseQuery, abstractSession, databaseMetadata);
```

Comment:
This method seems to get the hostname (defaulting to `"database"` if it can’t) and update the `databaseMetadata` map, which I found hard to follow.

If you take up the rest of my suggestions in the PR, we could replace these two lines with this:

```java
Optional<Connection> maybeConnection = Optional.ofNullable(abstractSession.getAccessor()).map(Accessor::getConnection);
Map<String, Object> databaseMetadata = maybeConnection.map(connection -> getMetadata(databaseQuery, connection)).orElseGet(HashMap::new);
String hostname = maybeConnection.flatMap(this::getDatabaseHostName).orElse("database");
```